File: 190.ReverseBits/reverse.py
class Solution:
    # @param n, an integer
    # @return an integer
    def reverseBits(self, n):

        bits = [0] * 32
        m = 0
        while n > 0:
            bits[m] = n % 2
            n /= 2
            m += 1

        print(bits)

        res = 0
        for i in range(len(bits)):
            res = res * 2 + bits[i]

        return res

if __name__ == '__main__':
    print(Solution().reverseBits(2147483648))

File: 190.ReverseBits/reverse.cc
class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        int res = 0;
        for (int i = 0; i < 32; i++) {
            res += n & 1;
            n >>= 1;
            if (i < 31) {
                res <<= 1;
            }
        }

        return res;
    }
};
